### DISCUSSION TITLE
Map ID, Version or Workshop id from demo

### DISCUSSION THREAD

**Discussion by IOExceptional** [saul/demofile#268] [Questions] [answered]
I've taken a look through some of the source here but not really seen it. 

Is it possible to get the map id/version or workshop id for a given map?

With the default maps, they can be deleted from the game files and then opening a demo will redownload them so I'm presuming something might be baked into the demo to figure out what version of the map was played

**Comment by IOExceptional** [comment]
Just noticed #158 asked a very similar thing, doesn't look like the default maps have this capability

**Comment by saul** [comment, marked as answer]
The official game client keeps a mapping of historic map CRC -> workshop ID. For example, if a demo was recorded in `de_inferno` with CRC `12345678`, it knows that it can download it from the workshop with ID `87654321`. This is based on a hardcoded list in the game client. This hardcoded list is referenced in the engine DLL here: https://github.com/SteamDatabase/GameTracking-CSGO/blob/master/bin/engine_client_strings.txt#L20172

Unfortunately I don't know have a way to extract this hardcoded list from the game client.

**Comment by IOExceptional** [comment]
That's incredibly helpful, thanks very much!

**Comment by markus-wa** [comment]
:wave: all - this has been something I've been personally battling with for quite a while now but I've finally figured out a way - I thought this might still be useful to someone here.

[saiko-tech/csgo-centrifuge](https://github.com/saiko-tech/csgo-centrifuge) now enables getting the correct radar image for any map version.

The core library + CLI to achieve this is open source and usage is explained in the repo README, but the easiest is to use the free public service.

### Usage

#### URL Formats
- `https://radar-overviews.csgo.saiko.tech/<map>/<crc>/radar.dds`
- `https://radar-overviews.csgo.saiko.tech/<map>/<crc>/radar.png`
- `https://radar-overviews.csgo.saiko.tech/<map>/<crc>/info.txt` - VDF format (Valve Data Format)

#### Examples

- https://radar-overviews.csgo.saiko.tech/cs_agency/2230463619/radar.dds
- https://radar-overviews.csgo.saiko.tech/cs_agency/2230463619/radar.png
- https://radar-overviews.csgo.saiko.tech/cs_agency/2230463619/info.txt

:warning: Currently the very latest map versions are not supported - so you need to use a fall-back image in case the requested radar image is missing (i.e. when you get a HTTP 404 error)!
